aspnetcore/blazor/fundamentals/render-modes.md

```diff
@@ -0,0 +1,295 @@
+---
+title: ASP.NET Core Blazor render modes
+author: guardrex
+description: Learn about Blazor render modes and how to apply them in Blazor Web Apps.
+monikerRange: '>= aspnetcore-8.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 09/08/2023
+uid: blazor/fundamentals/render-modes
+---
+# ASP.NET Core Blazor render modes
+
+This article explains how to establish hosting models for rendered Razor components in Blazor Web Apps, either at compile time or runtime.
+
+## Render modes
+
+Every component in a Blazor Web App adopts a *render mode* to determine the hosting model that it uses, where it's rendered, and whether or not it's interactive with C# code.
+
+The following table shows the four potential render mode scenarios for rendering Razor components in a Blazor Web App. The render mode attributes in the *Render mode attribute* column are applied to components with the [`@attribute` Razor directive](xref:mvc/views/razor#attribute). Later in this article, examples are shown for each render mode scenario.
+
+Scenario                      | Render mode attribute     | Hosting model                             | Render location     | Interactive
+----------------------------- | :-----------------------: | :---------------------------------------: | :-----------------: | :---------:
+Statically-rendered           | None                      | None                                      | Server              | ❌
+*Server-side rendering (SSR)* | `[RenderModeServer]`      | Blazor Server                             | Server              | ✔️
```

Comment:
This is different than how most other frontend web frameworks refer to the SSR. Typically SSR means the component is rendered statically in response to a request. For us, the Server render model means Blazor Server, which is connected, stateful, and interactive.

Reply:
I do agree though that Blazor Server is a form of server-side rendering, which is the source of confusion you pointed out in our email discussion. We're considering if we should rename the Server render mode to InteractiveServer and if we should add the same Interactive prefix to the other interactive render modes.

Reply:
I suppose the changes are to ...

* Server render mode
* WebAssembly render mode

... and update the headings later in the doc to match.